applications/GeoMechanicsApplication/custom_elements/interface_element.cpp

```diff
@@ -323,12 +333,71 @@ std::vector<Vector> InterfaceElement::CalculateTractionsAtIntegrationPoints(cons
 void InterfaceElement::ApplyRotationToBMatrix(Matrix& rBMatrix, const Matrix& rRotationMatrix) const
 {
     const auto dim = GetGeometry().WorkingSpaceDimension();
-    for (int i = 0; i + dim <= rBMatrix.size2(); i += dim) {
+    for (auto i = std::size_t{0}; i + dim <= rBMatrix.size2(); i += dim) {
         auto sub_matrix = subrange(rBMatrix, 0, rBMatrix.size1(), i, i + dim);
         sub_matrix.assign(Matrix{prod(sub_matrix, trans(rRotationMatrix))});
     }
 }
 
+void InterfaceElement::InterpolateNodalStressesToIntegrationPointTractions(
+    const std::vector<std::optional<Vector>>& interface_nodal_cauchy_stresses) const
+{
+    // interpolate nodal stresses on a chosen side
+    auto&      r_interface_geometry = GetGeometry();
+    const auto number_of_nodes_on_side =
+        GeometryUtilities::GetNodeIdsFromGeometry(r_interface_geometry).size() / 2;
+    // check which side is shared with the neighbour geometry, by checking for the first nodes of the first side
+    auto in_first_side = interface_nodal_cauchy_stresses[0].has_value();
+
+    const std::size_t   start_index = in_first_side ? 0 : number_of_nodes_on_side;
+    std::vector<Vector> nodal_stresses;
+    for (auto i = 0; i < number_of_nodes_on_side; ++i) {
+        nodal_stresses.push_back(interface_nodal_cauchy_stresses[start_index + i].value());
+    }
+    KRATOS_DEBUG_ERROR_IF(nodal_stresses.size() < number_of_nodes_on_side)
+        << "Less stresses found than needed for 1 side of the interface element with Id."
+        << this->Id() << std::endl;
+
+    std::size_t integration_point_index = 0;
+    for (const auto& r_integration_point : mIntegrationScheme->GetIntegrationPoints()) {
+        // interpolate on interface
+        auto integration_point_stress                = Vector{ZeroVector(nodal_stresses[0].size())};
+        auto integration_point_shape_function_values = Vector{};
+        r_interface_geometry.ShapeFunctionsValues(integration_point_shape_function_values, r_integration_point);
+        for (std::size_t i = 0; i < GetGeometry().PointsNumber() / 2; ++i) {
+            integration_point_stress += integration_point_shape_function_values[i] * nodal_stresses[i];
+        }
+
+        auto rotation_matrix = Matrix(3, 3, 0.0);
+        if (r_interface_geometry.LocalSpaceDimension() == 1) {
+            auto two_d_rotation_matrix = mfpCalculateRotationMatrix(GetGeometry(), r_integration_point);
+            GeoElementUtilities::AssembleUUBlockMatrix(rotation_matrix, two_d_rotation_matrix);
+            rotation_matrix(2, 2) = 1.0;
+        } else {
+            rotation_matrix = mfpCalculateRotationMatrix(GetGeometry(), r_integration_point);
+        }
+
+        auto integration_point_stress_tensor = MathUtils<double>::StressVectorToTensor(integration_point_stress);
```

Comment:
There is no need to supply a value for the template parameter here. It has a default value, which happens to be the one that you need:
```suggestion
        auto integration_point_stress_tensor = MathUtils<>::StressVectorToTensor(integration_point_stress);
```

Reply:
done